consumption.tex

```diff
@@ -0,0 +1,50 @@
+
+The transaction inclusion proof is conveniently defined by both Bitcoin and Ethereum. In Bitcoin, each block header contains a commitment to the set of the transaction ids in the block in the form of a Merkle Tree root. Thus, for Bitcoin our transaction inclusion proof is of the form $\mathsf{txInclusion} = (\mathsf{txIDMerkleRoot}, \mathsf{txIDMerkleProof})$. The shape of the proof is as follows $\mathsf{txIDMerkleProof} = ([x_0, x_1, \dots], \mathsf{txIndex}, \mathsf{txCount})$.
+
+Ethereum stores a similar commitment in its header, in the form of the root of a Patricia-Merkle Trie. This trie contains the transactions of the block, using the transaction id of each transaction as a key to determine the path to follow starting from the root in order to arrive at that transaction. Similar to regular Merkle trees, Patricia-Merkle tries allow for leaf inclusion proofs. Thus the form of \textsf{txInclusion} is the same as for Bitcoin transaction inclusion proofs. However, the shape of the merkle proof is different: $\mathsf{txIDMerkleProof} = [[x_{1,1}, x_{1,2}, \dots], \dots]$.
+
+The verification algorithm for $\alpha$ is presented in Algorithm~\ref{alg.verify-tx}. At the heart of the verification is the function \textsf{containsTransfer}, which takes a transaction, an address and an amount and determines if a value transfer of that exact amount to the address provided has ocured inside the transaction provided. For Bitcoin, this function locates a P2PKH output with the exact amount and receiving address. For Ethereum, the function ensures that the transaction comprises a fund transfer (i.e. the \textsf{data} field is empty), to the receiving address provided and with the exact amount. If this check passes and the transaction inclusion proofs are valid, \textsf{verify-tx} function accepts the proof $\alpha$ for the specified event $e$.
+
+\subsection{Block Connection Proof}
+
+We will look at two ways of proving block connection to the honestly adopted source chain.
+
+\subsubsection{Federation}
+A federation of $m$ nodes is picked. A federation node is responsible for running the source blockchain software and knowing at every moment the honestly adopted source blockchain. More importantly, it is responsible for transmitting this information to the target blockchain. In order to do this, we define a smart contract on the target blockchain where federation nodes are allowed to submit such information. We call this contract the \textsf{TagRepository}. A federation node encodes the honestly adopted chain into a tag and submits this tag to the \textsf{TagRepository}. Once a majority of votes is achieved, the tag is considered valid for use.
+
+Similar to FlyClient~\cite{flyclient}, we propose encoding the honestly adopted chain by using a Merkle Mountain Range (MMR). Specifically, to generate a tag we take all the chain's block ids in sequence and use them as leaves in a Merkle Mountain Range. We then use the root of that MMR as our encoding.
```

Comment:
The comparison to FlyClient can be moved to the related work section. Here we can just cite them as a reference for MMRs, but without naming them: `We propose encoding the honestly adopted chain using Merkle Mountain Ranges (MMRs)~\cite{flyclient}`.